c_emulator/riscv_callbacks.cpp

```diff
@@ -1,139 +1,99 @@
+#include "riscv_callbacks_if.h"
 #include "riscv_callbacks.h"
-#include "riscv_config.h"
-#include "riscv_platform_impl.h"
-#include "riscv_sail.h"
-#include <stdlib.h>
-#include <vector>
-#include <inttypes.h>
+#include <set>
 
-void print_lbits_hex(lbits val, int length = 0)
+std::set<callbacks_if *> callbacks;
```

Comment:
```suggestion
std::vector<callbacks_if *> callbacks;
```
I don't think there is any need for this to be a std::set which will make iteration over the elements slower.